### PR TITLE
Pointrend guide

### PR DESCRIPTION
**Pull request by kapil-varshney**
Updated guides for the following segmentation models with PointRend Enhancement:
1. PSPNet
2. MaskRCNN
3. DeepLabV3

Added YOLOv3 guide

### CODE REVIEW

**Comment by priyankatuteja** [comment]
@guneetmutreja Can you please review yolo guide, @vikashranjan26 Please check the pointrend part

File: guide/14-deep-learning/How_MaskRCNN_works.ipynb [hunk 1]
```diff
@@ -4,7 +4,7 @@
    "cell_type": "markdown",
```
Comment (vikashranjan26) [inline guide/14-deep-learning/How_MaskRCNN_works.ipynb:4]: <p>Please remove this sentence 'An iterative subdivision algorithm at selected locations is used to make point-based segmentation predictions' instead we should write 'Image segmentation methods often predict labels on a low-resolution regular grid, e.g., 1/8-th of the input. To make it actual size of an input image, interpolation takes place while in pointrend iterative subdivision algorithm is used to make it actual size of an image by predicting labels of the point by a trained small neural network'.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/881/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

**Review by vikashranjan26 — CHANGES_REQUESTED** [review]
@kapil-varshney Please incorporate minor suggested changes in pointrend and please correct the grammatical mistakes from my end if any.

File: guide/14-deep-learning/yolov3_object_detector.ipynb [hunk 2]
```diff
@@ -0,0 +1,153 @@
+{
```
Comment (guneetmutreja) [inline guide/14-deep-learning/yolov3_object_detector.ipynb:1]: <p>These models are already a part of arcgis python API change to ArcGIS API for Python.<strong> </strong></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/881/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (guneetmutreja) [inline guide/14-deep-learning/yolov3_object_detector.ipynb:1]: <p>.. concatenation of feature layers with earlier feature layers which <strong>preserve</strong> fine-grained features. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/881/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (guneetmutreja) [inline guide/14-deep-learning/yolov3_object_detector.ipynb:1]: <p>Remove this unwanted cell.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/881/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

File: guide/14-deep-learning/How_MaskRCNN_works.ipynb (hunk 1 above)
Comment (kapil-varshney) [reply guide/14-deep-learning/How_MaskRCNN_works.ipynb:4]: <p>Thanks! Updated.</p>

**Review by vikashranjan26 — APPROVED** [review]
LGTM

**Review by priyankatuteja — APPROVED** [review]

File: guide/14-deep-learning/How_MaskRCNN_works.ipynb (hunk 1 above)
Comment (jyaistMap) [inline guide/14-deep-learning/How_MaskRCNN_works.ipynb:4]: <p>The question mark should be removed unless the title is changed to "How Does MaskRCNN Work?"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/881/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (jyaistMap) [inline guide/14-deep-learning/How_MaskRCNN_works.ipynb:4]: <ul><li>Octothorpes (##) should be removed.</li></ul><p> </p><ul><li><em>over-smoothen&nbsp;</em>does not sound right in the first sentence. I think it sounds better to say:</li></ul><p>						<em>Segmentation models can tend to over generalize boundaries </em>								<em>which might not be precise for objects or scenes with irregular boundaries.</em></p><p>Or:</p><p>						<em>Segmentation models can tend to remove to many details from boundaries which might not be precise for objects or scenes with irregular boundaries.</em></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/881/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

File: guide/14-deep-learning/how_deeplabv3_works.ipynb [hunk 3]
```diff
@@ -4,14 +4,14 @@
    "cell_type": "markdown",
```
Comment (jyaistMap) [inline guide/14-deep-learning/how_deeplabv3_works.ipynb:4]: <p><em>over-smoothen&nbsp;</em>does not sound right in the first sentence. I think it sounds better to say:</p><p><em>Segmentation models can tend to over generalize boundaries which might not be precise for objects or scenes with irregular boundaries.</em></p><p>Or:</p><p><em>Segmentation models can tend to remove to many details from boundaries which might not be precise for objects or scenes with irregular boundaries.</em></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/881/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

File: guide/14-deep-learning/how_pspnet_works.ipynb [hunk 4]
```diff
@@ -260,7 +260,7 @@
    "cell_type": "markdown",
```
Comment (jyaistMap) [inline guide/14-deep-learning/how_pspnet_works.ipynb:260]: <p><em>over-smoothen </em>does not sound right in the first sentence. I think it sounds better to say:</p><p>     <em>Segmentation models can tend to over generalize boundaries which might not be precise for objects or scenes with irregular boundaries.</em></p><p> </p><p>Or:</p><p>    <em>Segmentation models can tend to remove to many details from boundaries which might not be precise for objects or scenes with irregular boundaries.</em></p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/881/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

File: guide/14-deep-learning/yolov3_object_detector.ipynb [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
```
Comment (jyaistMap) [inline guide/14-deep-learning/yolov3_object_detector.ipynb:1]: <p>This sentence should be edited to:</p><p><em>Adapting the ResNet style residual layers has improved its accuracy while maintaining the speed advantage.</em></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/881/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

**Review by jyaistMap — CHANGES_REQUESTED** [review]
Added some minor typo/grammar suggestions. Content looks good.

File: guide/14-deep-learning/How_MaskRCNN_works.ipynb (hunk 1 above)
Comment (kapil-varshney) [reply guide/14-deep-learning/How_MaskRCNN_works.ipynb:4]: <p>This is consistent with all the other guides.</p>

File: guide/14-deep-learning/yolov3_object_detector.ipynb (hunk 5 above)
Comment (kapil-varshney) [reply guide/14-deep-learning/yolov3_object_detector.ipynb:1]: <p>Edited. Thanks.</p>

File: guide/14-deep-learning/How_MaskRCNN_works.ipynb (hunk 1 above)
Comment (kapil-varshney) [reply guide/14-deep-learning/How_MaskRCNN_works.ipynb:4]: <p>I've moved the "PointRend Enhancement" to a different cell. Keeping the (###) to format it as a sub-heading.</p>

Comment (kapil-varshney) [reply guide/14-deep-learning/How_MaskRCNN_works.ipynb:4]: <p>"Smooth boundaries" is a common term in this context and has a meaning that "generalize boundaries" doesn't convey. "Over" is used to convey that it is more than what it needs to be. The abstract (https://arxiv.org/abs/1912.08193) uses "...boundaries in regions that are <em>over-smoothed</em> by previous methods."</p><p> </p><p>Changed the sentence to <em>Segmentation models can tend to generate over-smooth boundaries which might not be precise for objects or scenes with irregular boundaries.</em></p>

File: guide/14-deep-learning/how_deeplabv3_works.ipynb (hunk 3 above)
Comment (kapil-varshney) [reply guide/14-deep-learning/how_deeplabv3_works.ipynb:4]: <p>"Smooth boundaries" is a common term in this context and has a meaning that "generalize boundaries" doesn't convey. "Over" is used to convey that it is more than what it needs to be. The abstract (https://arxiv.org/abs/1912.08193) uses "...boundaries in regions that are <em>over-smoothed</em> by previous methods."</p><p> </p><p>Changed the sentence to <em>Segmentation models can tend to generate over-smooth boundaries which might not be precise for objects or scenes with irregular boundaries.</em></p>

File: guide/14-deep-learning/how_pspnet_works.ipynb (hunk 4 above)
Comment (kapil-varshney) [reply guide/14-deep-learning/how_pspnet_works.ipynb:260]: <p>"Smooth boundaries" is a common term in this context and has a meaning that "generalize boundaries" doesn't convey. "Over" is used to convey that it is more than what it needs to be. The abstract (https://arxiv.org/abs/1912.08193) uses "...boundaries in regions that are <em>over-smoothed</em> by previous methods."</p><p> </p><p>Changed the sentence to <em>Segmentation models can tend to generate over-smooth boundaries which might not be precise for objects or scenes with irregular boundaries.</em></p>

**Review by AtmaMani — APPROVED** [review]

**Pull request merged**